Doc/library/sqlite3.rst

```diff
@@ -2358,6 +2340,111 @@ can be found in the `SQLite URI documentation`_.
 .. _SQLite URI documentation: https://www.sqlite.org/uri.html
 
 
+.. _sqlite3-howto-row-factory:
+
+How to create and use row factories
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+By default, :mod:`!sqlite3` represent fetched rows as :class:`tuples <tuple>`.
+If a :class:`!tuple` does not suit your needs,
+use the :class:`sqlite3.Row` class or a custom :attr:`~Connection.row_factory`.
+
+:class:`!Row` provides indexed and case-insensitive named access to columns,
+with low memory overhead and minimal performance impact.
+In order to use :class:`Row` as a row factory,
+simply assign it to the :attr:`Connection.row_factory` attribute:
+
+.. doctest::
+
+   >>> con = sqlite3.connect(":memory:")
+   >>> con.row_factory = sqlite3.Row
+
+Query results are now processed using :class:`Row`:
+
+.. doctest::
+
+   >>> res = con.execute("SELECT 'Earth' AS name, 6378 AS radius")
+   >>> row = res.fetchone()
+   >>> row.keys()
+   ['name', 'radius']
+   >>> row[0]  # Access by index.
+   'Earth'
+   >>> row["name"]    # Access by name.
+   'Earth'
+   >>> row["RADIUS"]  # Column names are case-insensitive.
+   6378
+
+It is also possible to assign row factories to cursors using
+:attr:`Cursor.row_factory`:
+
+.. doctest::
+
+   >>> cur = con.cursor()
+   >>> cur.row_factory == con.row_factory
+   True
+   >>> cur.row_factory = None  # Override cursor row factory.
+
+   # The cursor and the connection row factories now differ.
+   >>> cur.row_factory == con.row_factory
+   False
+   >>> cur.execute("SELECT 'Hello'").fetchone()
+   ('Hello',)
+
+   # The connection still uses sqlite3.Row, as set in the previous example.
+   >>> con.execute("SELECT 'Hello'").fetchone()
+   <sqlite3.Row object at ...>
+
+If more flexibility is needed, implement a custom row factory.
+Here's an example of one returning a :class:`dict`:
```

Comment:
```suggestion
The following example shows how to return a :class:`dict`:
```

Avoid repeating the same phrasing twice in consecutive introduction lines

Reply:
IMO «example of one [row factory] returning a dict» is more concrete and understandable than «how to return a dict»

Reply:
Okay, sure; I wasn't 100% solid on this one. I'll suggest a change to the other one instead.

Reply:
I'm fine with either, and I think the messages gets through in both variations. If both of you care strongly for one of these, I'll gladly change it.